llvm/include/llvm/CodeGen/MachineOperand.h

```diff
@@ -446,6 +446,40 @@ class MachineOperand {
     return IsDebug;
   }
 
+  /// Whether this operand could be a sealed capability.
+  /// Note: this is very conservative and callers should look at the
+  /// instructions definition for register operands.
+  bool maybeSealed() const {
+    if (isReg())
+      return true; // Note: this is very conservative.
+    if (isFI())
+      return false; // FrameIndexes are always unsealed.
+    if (isCPI() || isTargetIndex() || isGlobal() || isBlockAddress() || isJTI())
+      return true; // Addresses of globals could be sealed.
+    llvm_unreachable("Wrong MachineOperand accessor for maybeSealed");
+    return true;
+  }
+  /// Whether this operand could be an untagged capability.
+  /// Note: this is very conservative and callers should look at the
+  /// instructions definition for register operands.
+  bool maybeUntagged() const {
+    if (isReg())
+      return true;
+    if (isFI() || isBlockAddress())
+      return false; // FrameIndexes and block addresses are always tagged.
+    if (isCPI() || isTargetIndex() || isGlobal() || isJTI())
+      return false;
+    if (isSymbol() || isMCSymbol()) {
+      // TODO: what about these? Are they always valid capabilities? Can they be
+      // used as operands for CHERI instructions? I don't think so, but let's
+      // add an explicit assertion message.
```

Comment:
I have a feeling these might get used for some libcalls. If so, I don't recall if they show up as being fat pointers yet, and they _tend_ to only be used for call instructions (well, CLGC and then indirect call) so might be ok?